<?xml version="1.0"?>

<launch>
  <env name="GAZEBO_MODEL_PATH" value="$(find igvc_description)"/>
  <arg name="world_name" default="$(find igvc_description)/urdf/igvc.world"/>
  <arg name="cmd_timeout" default="0.5"/>


  <!-- Vehicle pose -->
  <arg name="x" default="0"/>
  <arg name="y" default="0"/>
  <arg name="z" default="0"/>
  <arg name="roll" default="0.0"/>
  <arg name="pitch" default="0.0"/>
  <arg name="yaw" default="0"/>

  <node name="waypoint_source" pkg="igvc_navigation" type="waypoint_source" output="screen" >
      <param name="file" value="$(find igvc_gazebo)/config/waypoints.csv" />
  </node>

  <!-- Create the world. -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)"/> <!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- needed for rviz -->
  <param name="robot_description" command="cat $(find igvc_description)/urdf/jessi.urdf" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

  <!-- convert laser scan to point cloud for LIDAR -->
  <node name="scan_to_pointcloud" pkg="igvc_gazebo" type="scan_to_pointcloud" output="screen" >
    <param name="min_dist" value="0.1"/>
    <param name="neighbor_dist" value="0.2"/>
    <param name="offset" value="0"/>
  </node>
  <include file="$(find igvc_description)/launch/spawn_jaymii.launch" />
  <include file="$(find igvc_perception)/launch/filter_lidar.launch" />

  <include file="$(find igvc_gazebo)/launch/igvc_control.launch" />

</launch>
